src/Symfony/Component/Serializer/Normalizer/ConstraintViolationListNormalizer.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Serializer\Normalizer;
+
+use Symfony\Component\Validator\ConstraintViolationListInterface;
+
+/**
+ * A normalizer that normalize a ConstraintViolationListInterface instance.
+ *
+ * This Normalizer implements RFC7807 {@link https://tools.ietf.org/html/rfc7807}.
+ *
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ * @author Kévin Dunglas <dunglas@gmail.com>
+ */
+class ConstraintViolationListNormalizer implements NormalizerInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function normalize($object, $format = null, array $context = array())
+    {
+        $violations = array();
+        $messages = array();
+        foreach ($object as $violation) {
+            $violations[] = array(
+                'property_path' => $violation->getPropertyPath(),
+                'message' => $violation->getMessage(),
+                'code' => $violation->getCode(),
+            );
+            $propertyPath = $violation->getPropertyPath();
+            $prefix = $propertyPath ? sprintf('%s: ', $propertyPath) : '';
+            $messages[] = $prefix.$violation->getMessage();
+        }
+
+        return array(
+            'type' => isset($context['type']) ? $context['type'] : 'https://tools.ietf.org/html/rfc2616#section-10',
+            'title' => isset($context['title']) ? $context['title'] : 'An error occurred',
+            'detail' => $messages ? implode("\n", $messages) : (string) $object,
```

Comment:
Not sure we can safely rely on `(string) $object` (`__toString` is not part of the interface). Do we really need the fallback?